src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -407,5 +432,29 @@ private Task ProcessNegotiate(HttpContext context)
 
             return connectionState;
         }
+
+        private IList<Message> ParseSendBatch(ref BytesReader payload, MessageFormat messageFormat)
+        {
+            var messages = new List<Message>();
+
+            if (payload.Unread.Length == 0)
+            {
+                return messages;
+            }
+
+            if (payload.Unread[0] != MessageFormatter.GetFormatIndicator(messageFormat))
+            {
+                throw new FormatException($"Format indicator '{(char)payload.Unread[0]}' does not match format determined by Content-Type '{MessageFormatter.GetContentType(messageFormat)}'");
+            }
+
+            payload.Advance(1);
+
+            _parser.Reset();
```

Comment:
Yes, good catch. In the transports it's OK because each transport instance is per-connection